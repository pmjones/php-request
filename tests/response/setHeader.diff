001+ bool(false)
001- bool(true)
006+ 
007+ Fatal error: Uncaught Error: Call to undefined method SapiResponse::unsetHeaders() in /Users/pmjones/Code/pmjones/php-request/tests/response/setHeader.php:6
008+ Stack trace:
009+ #0 {main}
010+   thrown in /Users/pmjones/Code/pmjones/php-request/tests/response/setHeader.php on line 6
006- NULL
007- array(2) {
008-   ["foo-bar"]=>
009-   string(3) "baz"
010-   ["dib-zim"]=>
011-   string(3) "gir"
012- }
013- array(1) {
014-   ["dib-zim"]=>
015-   string(3) "gir"
016- }