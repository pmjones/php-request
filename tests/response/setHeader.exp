bool(true)
array(1) {
  ["foo-bar"]=>
  string(3) "dib"
}
NULL
array(2) {
  ["foo-bar"]=>
  string(3) "baz"
  ["dib-zim"]=>
  string(3) "gir"
}
array(1) {
  ["dib-zim"]=>
  string(3) "gir"
}