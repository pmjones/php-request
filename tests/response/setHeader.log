
---- EXPECTED OUTPUT
bool(true)
array(1) {
  ["foo-bar"]=>
  string(3) "dib"
}
NULL
array(2) {
  ["foo-bar"]=>
  string(3) "baz"
  ["dib-zim"]=>
  string(3) "gir"
}
array(1) {
  ["dib-zim"]=>
  string(3) "gir"
}
---- ACTUAL OUTPUT
bool(false)
array(1) {
  ["foo-bar"]=>
  string(3) "dib"
}

Fatal error: Uncaught Error: Call to undefined method SapiResponse::unsetHeaders() in /Users/pmjones/Code/pmjones/php-request/tests/response/setHeader.php:6
Stack trace:
#0 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/tests/response/setHeader.php on line 6
---- FAILED
