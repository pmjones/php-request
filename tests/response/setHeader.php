<?php
$response = new SapiResponse();
var_dump($response->setHeader('Foo-Bar', 'baz') === $response);
$response->setHeader('foo-bar', 'dib');
var_dump($response->getHeaders());
$response->unsetHeaders();
var_dump($response->getHeaders());
$response->setHeader('Foo-Bar', 'baz');
$response->setHeader('dib-zim', 'gir');
var_dump($response->getHeaders());
$response->unsetHeader('foo-bar');
$response->unsetHeader('no-such');
var_dump($response->getHeaders());
