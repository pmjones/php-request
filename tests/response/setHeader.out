bool(false)
array(1) {
  ["foo-bar"]=>
  string(3) "dib"
}

Fatal error: Uncaught Error: Call to undefined method SapiResponse::unsetHeaders() in /Users/pmjones/Code/pmjones/php-request/tests/response/setHeader.php:6
Stack trace:
#0 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/tests/response/setHeader.php on line 6